src/algokit/cli/tasks/transfer.py

```diff
@@ -0,0 +1,194 @@
+
+    return asset_info.get("decimals", 0)
+
+
+def _validate_address(address: str) -> None:
+    if not algosdk.encoding.is_valid_address(address):  # type: ignore[no-untyped-call]
+        raise click.ClickException(f"{address} is an invalid account address")
+
+
+def _validate_asset_balance(sender_account_info: dict, receiver_account_info: dict, asset_id: int, amount: int) -> None:
+    sender_asset_record = next(
+        (asset for asset in sender_account_info.get("assets", []) if asset["asset-id"] == asset_id), None
+    )
+    receiver_asset_record = next(
+        (asset for asset in receiver_account_info.get("assets", []) if asset["asset-id"] == asset_id), None
+    )
+    if not sender_asset_record:
+        raise click.ClickException("Sender is not opted into the asset")
+    if sender_asset_record["amount"] < amount:
+        raise click.ClickException("Insufficient asset balance in sender account")
+    if not receiver_asset_record:
+        raise click.ClickException("Receiver is not opted into the asset")
+
+
+def _validate_balance(
+    sender: Account, receiver: str, asset_id: int, amount: int, algod_client: algosdk.v2client.algod.AlgodClient
+) -> None:
+    sender_account_info = algod_client.account_info(sender.address)
+    receiver_account_info = algod_client.account_info(receiver)
+
+    if not isinstance(sender_account_info, dict) or not isinstance(receiver_account_info, dict):
+        raise click.ClickException("Invalid account info response")
+
+    if asset_id == 0:
+        if isinstance(sender_account_info, dict) and sender_account_info.get("amount", 0) < amount:
+            raise click.ClickException("Insufficient Algos balance in sender account")
+    else:
+        _validate_asset_balance(sender_account_info, receiver_account_info, asset_id, amount)
+
+
+def _validate_inputs(
+    sender: Account, receiver: str, asset_id: int, amount: int, algod_client: algosdk.v2client.algod.AlgodClient
+) -> None:
+    _validate_address(receiver)
+    _validate_balance(sender, receiver, asset_id, amount, algod_client)
+
+
+def _get_sender_account(sender: str) -> Account:
+    try:
+        _validate_address(sender)
+        pk = _get_private_key_from_mnemonic()
+        return Account(address=sender, private_key=pk)
+    except click.ClickException as ex:
+        alias_data = get_alias(sender)
+
+        if not alias_data:
+            raise click.ClickException(f"Alias `{sender}` alias does not exist.") from ex
+        if not alias_data.private_key:
+            raise click.ClickException(f"Alias `{sender}` does not have a private key.") from ex
+
+        return Account(address=alias_data.address, private_key=alias_data.private_key)
+
+
+@click.command(name="transfer")
+@click.option("--sender", "-s", type=click.STRING, help="Address or alias of the sender account", required=True)
+@click.option(
+    "--receiver", "-r", type=click.STRING, help="Address to an account that will receive the asset(s)", required=True
```

Comment:
Assuming there was a conversation to not support aliases on receiver. The user story indicates that it should work with aliases.